ouroboros-network/test/Test/Ouroboros/Network/KeepAlive.hs

```diff
@@ -155,27 +155,23 @@ prop_keepAlive_convergenceM tracer (NetworkDelay nd) seed = do
     controlMessageV <- newTVarIO Continue
     let controlMessageSTM = readTVar controlMessageV
         clientId = "client"
-        timeConstant = 1000 -- Same as in PeerGSV's <> definition
         keepAliveInterval = 10
 
     (c_aid, s_aid) <- runKeepAliveClientAndServer (NetworkDelay nd) seed tracer controlMessageSTM
                           registry clientId (KeepAliveInterval keepAliveInterval)
-    threadDelay $ timeConstant * keepAliveInterval
 
-    atomically $ writeTVar controlMessageV Terminate
     void $ wait c_aid
     void $ wait s_aid
 
-    -- XXX Must be larger than the KeepAliveInterval timeout or we leak threads in the SIM
-    -- Can be removed after #2631 is merged.
-    threadDelay (keepAliveInterval + 128)
-
 -- Test that our estimate of PeerGSV's G terms converge to
 -- a given constant delay.
 prop_keepAlive_convergence :: NetworkDelay -> Int -> Property
 prop_keepAlive_convergence nd seed =
-    let trace = selectTraceEventsDynamic $ runSimTrace $ prop_keepAlive_convergenceM dynamicTracer nd seed in
-    verifyConvergence trace
+    let trace = take 1000
```

Comment:
We used to wait `1000s` now we take first `1000` traces.  Any idea how they correspond to each other?

Reply:
That's fine, I just wonder how much time moves in 1000 trace events.

Reply:
I am not sure, I just put a big number to increase the amount of events we analyze so we can be more confident in the test. I think 1000s was rather arbitrary as well.

Reply:
It's actually around `10000s`
```
  KeepAlive
    KeepAlive Convergence: OK (5.32s)
      +++ OK, passed 100 tests:
      15% 10600s
      12% 10400s
      12% 10700s
      10% 10200s
      10% 10500s
      10% 10800s
       9% 10000s
       8% 10900s
       7% 10100s
       6% 10300s
       1% 9900s
```

Reply:
Thank you!